plugins/azure-devops-backend/src/service/router.test.ts

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { getVoidLogger } from '@backstage/backend-common';
+import { ConfigReader } from '@backstage/config';
+import express from 'express';
+import request from 'supertest';
+import { AzureDevOpsApi } from '../api';
+import { createRouter } from './router';
+import { RepoBuild } from '../api/types';
+import { GitRepository } from 'azure-devops-node-api/interfaces/GitInterfaces';
+import {
+  Build,
+  BuildResult,
+  BuildStatus,
+} from 'azure-devops-node-api/interfaces/BuildInterfaces';
+
+describe('createRouter', () => {
+  let azureDevOpsApi: jest.Mocked<AzureDevOpsApi>;
+  let app: express.Express;
+
+  beforeAll(async () => {
+    azureDevOpsApi = {
+      getGitRepository: jest.fn(),
+      getBuildList: jest.fn(),
+      getRepoBuilds: jest.fn(),
+    } as any;
+    const router = await createRouter({
+      azureDevOpsApi,
+      logger: getVoidLogger(),
+      config: new ConfigReader({
+        azureDevOps: {
+          token: 'foo',
+          host: 'host.com',
+          organization: 'myOrg',
+          top: 5,
+        },
+      }),
+    });
+    app = express().use(router);
+  });
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('GET /health', () => {
+    it('returns ok', async () => {
+      const response = await request(app).get('/health');
+
+      expect(response.status).toEqual(200);
+      expect(response.body).toEqual({
+        status: 'Healthy',
+        details: 'All required config has been provided',
+      });
+    });
+  });
+
+  describe('GET /repository/:projectName/:repoName', () => {
```

Comment:
alternative perhaps a bit more RESTy path suggestion: `/projects/:projectName/repos/:repoName/{repositories,builds}`

(definitely a matter of taste, feel free to ignore)

Reply:
Yeah, I understand what you are suggesting. I'm mostly trying to mimic the current Azure DevOps API so that way if someone has issues and needs to work this back things line up